integrationTests/chainSimulator/mempool/mempool_test.go

```diff
@@ -485,3 +486,874 @@ func TestMempoolWithChainSimulator_Eviction(t *testing.T) {
 	expectedNumTransactionsInPool := 300_000 + 1 + 1 - int(storage.TxPoolSourceMeNumItemsToPreemptivelyEvict)
 	require.Equal(t, expectedNumTransactionsInPool, getNumTransactionsInPool(simulator, shard))
 }
+
+func Test_Selection_ShouldNotSelectSameTransactionsWithSameSender(t *testing.T) {
+	t.Parallel()
+
+	host := txcachemocks.NewMempoolHostMock()
+	txpool, err := txcache.NewTxCache(txcache.ConfigSourceMe{
+		Name:                        "test",
+		NumChunks:                   16,
+		NumBytesThreshold:           maxNumBytesUpperBound,
+		NumBytesPerSenderThreshold:  maxNumBytesPerSenderUpperBoundTest,
+		CountThreshold:              math.MaxUint32,
+		CountPerSenderThreshold:     math.MaxUint32,
+		EvictionEnabled:             false,
+		NumItemsToPreemptivelyEvict: 1,
+		TxCacheBoundsConfig: config.TxCacheBoundsConfig{
+			MaxNumBytesPerSenderUpperBound: maxNumBytesPerSenderUpperBoundTest,
+		},
+	}, host)
+
+	require.Nil(t, err)
+	require.NotNil(t, txpool)
+
+	// create the non-virtual selection session, assure we have enough balance
+	selectionSession := createMockSelectionSession(map[string]*accountInfo{
+		"alice": {
+			balance: oneEGLD,
+			nonce:   0,
+		},
+		"receiver": {
+			balance: big.NewInt(0),
+			nonce:   0,
+		},
+	})
+
+	maxNumTxs := 2
+	options := holders.NewTxSelectionOptions(
+		10_000_000_000,
+		maxNumTxs,
+		selectionLoopMaximumDuration,
+		10,
+	)
+
+	poolTxs := maxNumTxs * 2
+	txHashes := make([][]byte, 0, poolTxs)
+
+	nonceTracker := newNoncesTracker()
```

Comment:
You can even simply drop the variable `poolTxs`, and discard the size hint for the slice (we are in a small test here).